lib/sqlalchemy/orm/mapper.py

```diff
@@ -1398,16 +1398,24 @@ def _configure_pks(self):
             self.primary_key = tuple(primary_key)
             self._log("Identified primary key columns: %s", primary_key)
 
-        # determine cols that aren't expressed within our tables; mark these
-        # as "read only" properties which are refreshed upon INSERT/UPDATE
+        # determine cols that are either table-derived expressions, or special
+        # table columns that generate a value and are never written towards.
+        # Mark these as "read only" properties which are refreshed upon
+        # INSERT/UPDATE and which are also candidates to block if a user-side
+        # value is set for them
         self._readonly_props = set(
             self._columntoproperty[col]
             for col in self._columntoproperty
             if self._columntoproperty[col] not in self._identity_key_props
```

Comment:
I'm not sure if we should manage the case of a computed column that is also primary key

Reply:
if we keep it this way, should there be a test for it? (ie a primary key will not warn even if computed)

Reply:
that would be very strange and I wonder if all the backends even allow the "GENERATED AS" on a primary key.  however, primary key cols not being "writeable" is a little more plausible at least, on SQL Server you can't easily write to an "identity" column in fact, the dialect does some kind of silly magic for that use case that if it were today I might not have it do.

that the identity_key_props are excluded here might indicate there's still some distinction between what "computed" is and what other kinds of column_property() attributes are, but we can always improve it later.

Reply:
I agree that this is an edge case. I think we could leave it this way and revisit it later it there is a need to.

Reply:
It seems that mssql allows some computed columns in the primary key. I'm not sure about the other backends. https://docs.microsoft.com/en-us/sql/relational-databases/tables/specify-computed-columns-in-a-table?view=sql-server-ver15